tools/tu_collector/tests/tu_collector_test.py

```diff
@@ -60,3 +60,36 @@ def test_file_existence(self):
         self.assertTrue(
             any([path.endswith(os.path.join('/', 'hello.c')) for path in files]))
         self.assertIn('compilation_database.json', files)
+
+    def test_ctu_collection(self):
+        ctu_deps_dir = tempfile.mkdtemp()
+
+        ctu_action = next(filter(lambda ba: ba['file'] == 'ctu.cpp',
+                                 self.compile_cmd_data))
+
+        hash_fun = dict(
+            inspect.getmembers(tu_collector))['__analyzer_action_hash']
+
+        with open(os.path.join(ctu_deps_dir, hash_fun(ctu_action)), 'w') as f:
+            f.write(os.path.join(self._test_proj_dir, 'zero.cpp'))
+
+        zip_file_name = tempfile.mkstemp(suffix='.zip')[1]
```

Comment:
Similar to my above comment. Use the following approach to create a temp file:
https://github.com/Ericsson/codechecker/blob/ed84de269ce575a23252175ffb0461757b747c45/web/tests/functional/cmdline/test_cmdline.py#L268